x-pack/plugins/security_solution/server/lib/detection_engine/schemas/rule_converters.ts

```diff
@@ -386,64 +386,69 @@ const shouldUpdateVersion = (params: PatchRulesSchema): boolean => {
-      riskScoreMapping: params.risk_score_mapping ?? existingParams.riskScoreMapping,
-      ruleNameOverride: params.rule_name_override ?? existingParams.ruleNameOverride,
-      setup: params.setup ?? existingParams.setup,
-      severity: params.severity ?? existingParams.severity,
-      severityMapping: params.severity_mapping ?? existingParams.severityMapping,
-      threat: params.threat ?? existingParams.threat,
-      timestampOverride: params.timestamp_override ?? existingParams.timestampOverride,
+      license: nextParams.license ?? existingParams.license,
+      outputIndex: nextParams.output_index ?? existingParams.outputIndex,
+      timelineId: nextParams.timeline_id ?? existingParams.timelineId,
+      timelineTitle: nextParams.timeline_title ?? existingParams.timelineTitle,
+      meta: nextParams.meta ?? existingParams.meta,
+      maxSignals: nextParams.max_signals ?? existingParams.maxSignals,
+      relatedIntegrations: nextParams.related_integrations ?? existingParams.relatedIntegrations,
+      requiredFields: nextParams.required_fields ?? existingParams.requiredFields,
+      riskScore: nextParams.risk_score ?? existingParams.riskScore,
+      riskScoreMapping: nextParams.risk_score_mapping ?? existingParams.riskScoreMapping,
+      ruleNameOverride: nextParams.rule_name_override ?? existingParams.ruleNameOverride,
+      setup: nextParams.setup ?? existingParams.setup,
+      severity: nextParams.severity ?? existingParams.severity,
+      severityMapping: nextParams.severity_mapping ?? existingParams.severityMapping,
+      threat: nextParams.threat ?? existingParams.threat,
+      timestampOverride: nextParams.timestamp_override ?? existingParams.timestampOverride,
       timestampOverrideFallbackDisabled:
-        params.timestamp_override_fallback_disabled ??
+        nextParams.timestamp_override_fallback_disabled ??
         existingParams.timestampOverrideFallbackDisabled,
-      to: params.to ?? existingParams.to,
-      references: params.references ?? existingParams.references,
-      namespace: params.namespace ?? existingParams.namespace,
-      note: params.note ?? existingParams.note,
+      to: nextParams.to ?? existingParams.to,
+      references: nextParams.references ?? existingParams.references,
+      namespace: nextParams.namespace ?? existingParams.namespace,
+      note: nextParams.note ?? existingParams.note,
       // Always use the version from the request if specified. If it isn't specified, leave immutable rules alone and
       // increment the version of mutable rules by 1.
-      version:
-        params.version ?? existingParams.immutable
-          ? existingParams.version
-          : shouldUpdateVersion(params)
-          ? existingParams.version + 1
-          : existingParams.version,
-      exceptionsList: params.exceptions_list ?? existingParams.exceptionsList,
+      version: nextParams.version
+        ? nextParams.version
+        : shouldUpdateVersion(nextParams)
```

Comment:
```suggestion
        : shouldUpdateVersion(nextParams) && !existingParams.immutable
```
I think we still want the rule converters to be aware of the rule's immutable flag to avoid automatically bumping the version on immutable rules when params change. The logic here was supposed to replace the logic in the old `calculateVersion` [function](https://github.com/elastic/kibana/pull/135479/files#diff-970e8ad0bda1495c981b78f5916aa064df1b40fa6734193aa54d1d479b0c1c9eL156) that I removed in https://github.com/elastic/kibana/pull/135479, but I put the `immutable` check in the wrong place :sweat:. (The old `calculateVersion` function would *only* use the provided version for immutable rules, not for mutable rules, which I think was a mistake - both mutable and immutable rules should use the version provided in the request, if specified, and if it's not then only mutable rules should auto-update the version).

If we don't include the immutable check then a user could patch a rule through the API, the version would automatically bump, and the next prebuilt rule update would not be applied to the rule.

Reply:
> If we don't include the immutable check then a user could patch a rule through the API, the version would automatically bump, and the next prebuilt rule update would not be applied to the rule.

Ohh, wow, I was under the impression that we do not allow users to modify `immutable` rules through the patch API, but it turned out to be false 😅 I just tried to send a couple of `PATCH api/detection_engine/rules` requests and was able to modify any prebuilt rule param.

Given that, I agree that we must keep the immutable check. Thanks for the heads up, @marshallmain.